source/AI.cpp

```diff
@@ -2631,35 +2618,19 @@ bool AI::DoCloak(Ship &ship, Command &command)
 
 
 bool AI::DoRoving(Ship &ship, Command &command, unsigned int rovingShipCount)
-{
-	const System* system = ship.GetSystem();
-	if((rovingShipCount > 9 && system))
-	{
-		if(ship.Fuel() >= ship.JumpFuel() || ship.Fuel() >= ship.JumpDriveFuel())
-		{
-			MoveIndependent(ship, command);
-			return true;
-		}
-		else if (system->IsInhabited(&ship))
-		{
-			for(const StellarObject& planet: system->Objects())
-				if(CanRefuel(ship, &planet))
-				{
-					ship.SetTargetStellar(&planet);
-				}
-			Refuel(ship, command);
-			return true;
-		}
-	}
-	
+{	
 	const Point target = ship.GetTargetPosition();
 	const auto v = ship.MaxVelocity();
-    if(!target || MoveTo(ship, command, target, Point(), v, v))
+	if(!target || MoveTo(ship, command, target, Point(), v, v))
     {
         Point newTarget = Angle::Random().Unit() * Random::Real() * MAX_DISTANCE_FROM_CENTER;
         ship.SetTargetPosition(newTarget);
     }
     
+    const System *system = ship.GetSystem();
+	if(rovingShipCount > 9 && system)
+		return DoBored(ship, command, system);
+    
```

Comment:
You forgot to fix the indentation here. Also, you can resolve all my comments :)

Reply:
Hm.